src/components/SigninModal.jsx

```diff
@@ -0,0 +1,78 @@
+import "../blocks/Modal.css";
+import { useContext } from "react";
+import ModalWithForm from "./ModalWithForm";
+import { AppContext } from "../contexts/AppContext";
+import { useFormValidation } from "../utils/useFormValidation";
+
+export default function SigninModal({ handleSignin, isVisible }) {
+  const { setActiveModal } = useContext(AppContext);
+  const { values, errors, isValid, handleChange, resetForm } =
+    useFormValidation();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!isValid) return;
+    handleSignin(values);
+    resetForm();
```

Comment:
You need to **reset** the form only when you **open** it or better in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.